### DISCUSSION TITLE
Getting intellisense to work in Razor files

### DISCUSSION THREAD

**Discussion by jerriep** [statiqdev/Discussions#79] [Q&A] [answered]
Is there a way to get IntelliSense to work in Razor files in Statiq projects?

The screenshot below is from the `statiqdev.github.io` project and as you can see it does not seem to recognise the `Outputs` property (or any of the other properties from the `StatiqRazorPage<TModel>` base class. Also, things like the `DocumentLink` HTML helper is not recognised or picked up in IntelliSense.

![image](https://user-images.githubusercontent.com/1006420/122406041-bef43f00-cfaa-11eb-8e6f-67c4ca9287e7.png)

This is the case for me in both VS and Rider. Perhaps there is a trick I am missing?

**Comment by daveaglick** [comment, marked as answer]
Yes! You're in luck, there's actually a pretty straight-forward way to let the Razor engine know that you're using a different base page class. Create a `_ViewImports.cshtml` file at the root of the "input" folder as described here: https://statiq.dev/web/content-and-data/templates/razor - let me know if you have any trouble.

**Reply by jerriep** [reply]
Hi Dave, it does not work. That screenshot above is from the Statiq website repo (https://github.com/statiqdev/statiqdev.github.io) which has that defined. Also on my own website which has it defined (https://github.com/stellar-admin/website/blob/develop/input/_viewimports.cshtml) this does not work. IntelliSense does not work and VS (and Rider) does not recognise the symbols like `Outputs`

**Reply by jerriep** [reply]
Ooh as I typed that response, I was wondering whether it was Jetbrains' fault, since I have ReSharper installed in VS. I disabled it quickly and now it works 100%. So, it seems it is ReSharper (and Rider) that does not like it.

I'll just use VS without Resharper for editing those templates. 

Thank you :)

**Reply by StevenTCramer** [reply]
@daveaglick  the link is broken.
Here is a permalink to @jerriep (thanks) current version 
https://github.com/stellar-admin/website/blob/dd8dfa2607d508752b139099bda1aa867b65f521/src/StellarAdminWebsite/input/_viewimports.cshtml

**Reply by StevenTCramer** [reply]
https://github.com/statiqdev/Website/blob/fb7f8f09cced081968588114492eb9df9c074dd5/input/guide/content-and-data/template-languages/razor.md

I found the doc Dave was referencing and above is a permalink.

**Reply by daveaglick** [reply]
@StevenTCramer I've been tweaking the guide layout to try and make the docs a little easier to follow. Of course, that also means I'm breaking links since it's shifting around enough that there's not always an obvious redirect 😬. Thanks a ton for permalinking that here for others who stumble on this same discussion.

**Reply by icalvo** [reply]
I've found a tweak that is a bit more general, because **it also works with Rider**. In addition to the `_ViewImport.cshtml` stuff, just change the project SDK. Edit the first line of your `.csproj` file to be:

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">
```

Now the intellisense works in Rider. I had to tune up the CleanBlog theme a little bit to get rid of all Razor complaints.
